### PR TITLE
feat: add Coordinates Broker and Subset

### PR DESCRIPTION
**Pull request by phc1990**

### CODE REVIEW

File: include/OpenSpaceToolkit/Astrodynamics/Trajectory/CoordinatesSubset.hpp [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+{
+
+using ostk::core::types::String;
+
+using ostk::astro::trajectory::CoordinatesSubset;
+
+template <>
+struct hash<CoordinatesSubset>
+{
+    size_t operator()(const CoordinatesSubset& aCoordinatesSubset) const
```
Comment (Derollez) [inline include/OpenSpaceToolkit/Astrodynamics/Trajectory/CoordinatesSubset.hpp:154]:
```suggestion
    Size operator()(const CoordinatesSubset& aCoordinatesSubset) const
```

Comment (Derollez) [reply include/OpenSpaceToolkit/Astrodynamics/Trajectory/CoordinatesSubset.hpp:154]: Do we leverage later?

File: src/OpenSpaceToolkit/Astrodynamics/Trajectory/CoordinatesBroker.cpp [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+}
+
+Size CoordinatesBroker::getNumberOfCoordinates() const
+{
+    return this->nextCoordinatesSubsetIndex_;
+}
+
+Size CoordinatesBroker::getNumberOfSubsets() const
+{
+    return (size_t)this->coordinatesSubsetsIndexMap_.size();
```
Comment (Derollez) [inline src/OpenSpaceToolkit/Astrodynamics/Trajectory/CoordinatesBroker.cpp:71]:
```suggestion
    return (Size)this->coordinatesSubsetsIndexMap_.size();
```

It's the same, simply aligning

File: src/OpenSpaceToolkit/Astrodynamics/Trajectory/CoordinatesBroker.cpp [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+{
+    return this->hasSubset(aCoordinatesSubsetSPtr->getId());
+}
+
+Index CoordinatesBroker::getSubsetIndex(const Shared<const CoordinatesSubset>& aCoordinatesSubsetSPtr) const
+{
+    return this->getSubsetIndex(aCoordinatesSubsetSPtr->getId());
+}
+
+VectorXd CoordinatesBroker::extract(const VectorXd& allCoordinates, const CoordinatesSubset& aCoordinatesSubset) const
```
Comment (Derollez) [inline src/OpenSpaceToolkit/Astrodynamics/Trajectory/CoordinatesBroker.cpp:107]: Should we be providing the `CoordinatesSubset` in input or its idea? 🤔

Comment (Derollez) [reply src/OpenSpaceToolkit/Astrodynamics/Trajectory/CoordinatesBroker.cpp:107]: Checking further in the PRs to get a better idea

**Comment by Derollez** [comment]
Should we move those concepts to `Trajectory > State` or create a new `Coordinates` folder under Trajectory? 🤔

File: include/OpenSpaceToolkit/Astrodynamics/Trajectory/CoordinatesSubset.hpp [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+    /// @param              [in] allCoordinates_1 first set of all coordinates
+    /// @param              [in] allCoordinates_2 second set of all coordinates
+    /// @param              [in] aFrame the reference frame in which the coordinates are resolved
+    /// @param              [in] aCoordinatesBroker a coordinates broker
+    ///
+    /// @return             The resulting coordinates subset value (subset_1 + subset_2)
+
+    VectorXd add(
+        const Instant& anInstant,
+        const VectorXd& allCoordinates_1,
```
Comment (Derollez) [inline include/OpenSpaceToolkit/Astrodynamics/Trajectory/CoordinatesSubset.hpp:92]:
```suggestion
        const VectorXd& allCoordinates_1,
```
I'd use `afullCoordinates` and `anotherFullCoordinates` to align

File: include/OpenSpaceToolkit/Astrodynamics/Trajectory/CoordinatesSubset.hpp (hunk 1 above)
Comment (phc1990) [reply include/OpenSpaceToolkit/Astrodynamics/Trajectory/CoordinatesSubset.hpp]: You're correct, this is a leftover! ty!

File: src/OpenSpaceToolkit/Astrodynamics/Trajectory/CoordinatesBroker.cpp (hunk 3 above)
Comment (phc1990) [reply src/OpenSpaceToolkit/Astrodynamics/Trajectory/CoordinatesBroker.cpp]: Sorry didn't ge the `or its idea` part

**Review by vishwa2710 — COMMENTED** [review]
First pass

**Review by vishwa2710 — APPROVED** [review]

File: src/OpenSpaceToolkit/Astrodynamics/Trajectory/CoordinatesSubset.cpp [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+    [[maybe_unused]] const VectorXd& allCoordinates_2,
+    [[maybe_unused]] const Shared<const Frame>& aFrame,
+    [[maybe_unused]] const Shared<const CoordinatesBroker>& aCoordinatesBroker
+) const
+{
+    throw ostk::core::error::runtime::ToBeImplemented("Addition");
+}
+
+VectorXd CoordinatesSubset::subtract(
+    [[maybe_unused]] const Instant& anInstant,
```
[the remark this answers is not in the document]
Comment (phc1990) [reply src/OpenSpaceToolkit/Astrodynamics/Trajectory/CoordinatesSubset.cpp]: @vishwa2710 looked it up "it's alright, just a more modern way to do (void)"

File: include/OpenSpaceToolkit/Astrodynamics/Trajectory/CoordinatesSubset.hpp [hunk 6]
```diff
@@ -0,0 +1,162 @@
...
+
+    bool operator!=(const CoordinatesSubset& aCoordinatesSubset) const;
+
+    /// @brief              Returns the unique identifier of the instance
+    ///
+    /// @return             The unique identifier of the instance
+
+    String getId() const;
+
+    /// @brief              Returns the name of the instance
```
[the remark this answers is not in the document]
Comment (phc1990) [reply include/OpenSpaceToolkit/Astrodynamics/Trajectory/CoordinatesSubset.hpp]: Done!

**Pull request merged**